libnet/configure.ac

```diff
@@ -4,25 +4,50 @@ dnl All rights reserved.
 dnl
 dnl Process this file with autoconf to produce a configure script.
 
-AC_INIT([libnet],[1.2-rc3])
+AC_INIT([libnet],[1.2-rc3],,,[http://github.com/sam-github/libnet])
 AC_MSG_RESULT(beginning autoconfiguration process for libnet-${PACKAGE_VERSION} ...)
 AC_CANONICAL_TARGET
 AC_CONFIG_SRCDIR([src/libnet_build_ip.c])
 AC_CONFIG_MACRO_DIR([m4])
-AM_INIT_AUTOMAKE
+AM_INIT_AUTOMAKE([foreign subdir-objects no-dependencies -Wall])
 AC_CONFIG_HEADERS([include/config.h])
-AM_MAINTAINER_MODE
 
 dnl
 dnl Check for the usual programs
 dnl
 AC_PROG_CC
 AC_PROG_INSTALL
 AC_USE_SYSTEM_EXTENSIONS
+AX_CFLAGS_WARN_ALL
 dnl disable shared libraries (for now)
 dnl AC_DISABLE_SHARED
+AM_PROG_AR
```

Comment:
The example on the info page doesn't cover the point I was trying to make.
You're right. By itself, foreign just disables some checks.
But in our case, we have this chain: If you don't set foreign, you'll need to set AM_PROG_AR. This macro (AM_PROG_AR) in turn, will (by default) abort configure with an error message if the archiver interface is not recognized (instead of just keep going), hence limiting portability.

I really can't think of an archiver that is too exotic for Automake. But since you don't have to have this macro (if you have foreign), I suggest you remove it.